### PR TITLE
allow users to disable X11 clipboard monitoring

### PR DESCRIPTION
**Pull request by kevinushey**
This provides an escape hatch for users being hit bit slowness when RStudio's clipboard monitoring enabled, as per https://github.com/rstudio/rstudio/issues/2068.

One can launch RStudio with something like:

```
RSTUDIO_NO_CLIPBOARD_MONITORING=1 rstudio
```

to activate this feature (pending extra UI for toggling this).

### CODE REVIEW

**Comment by kevinushey** [comment]
Issue tracking UI work is here: https://github.com/rstudio/rstudio/issues/2548

File: src/cpp/desktop/DesktopGwtCallback.cpp [hunk 1]
```diff
@@ -57,8 +57,13 @@ namespace desktop {
 
 namespace {
 WindowTracker s_windowTracker;
+
+#ifdef Q_OS_LINUX
 QString s_globalMouseSelection;
+bool s_clipboardMonitoringEnabled;
```
Comment (jmcphers) [inline src/cpp/desktop/DesktopGwtCallback.cpp]: Do we need a static here? (seems like it's recomputed on every access -- is this just a placeholder for a future preference-driven value?)

Comment (kevinushey) [reply src/cpp/desktop/DesktopGwtCallback.cpp]: Right; it's effectively just a placeholder. (Currently, we activate this only on RStudio launch; in theory we might want to activate / deactivate it on-the-fly in an active session)

Comment (jmcphers) [reply src/cpp/desktop/DesktopGwtCallback.cpp]: Sounds good!

**Pull request merged**